work/check_for_placeholders.py

```diff
@@ -0,0 +1,67 @@
+import argparse
+
+from lib.utils import *
+
+def run(site_id: str, org_id: str, app: dict):
+    logging.info('Replace placeholders with multimedia links for org_id: {}'.format(org_id))
+    site_folder = APP['archive_folder']
+
+    xml_src = r'{}{}-archive/lessonbuilder.xml'.format(site_folder, site_id)
+    xml_old = r'{}{}-archive/lessonbuilder.old'.format(site_folder, site_id)
+    shutil.copyfile(xml_src, xml_old)
+
+    remove_unwanted_characters(xml_src)
+
+    file_path = os.path.join(site_folder, xml_src)
+
+    with open(file_path, "r", encoding="utf8") as fp:
+        soup = BeautifulSoup(fp, 'xml')
+        pages = soup.find_all('page')
+        for page in pages:
+            items = page.find_all('item', attrs={"type": "5"})
+            for item in items:
+                html = BeautifulSoup(item.attrs['html'], 'html')
+                placeholder_found = html.find('p', attrs={"data-type": "placeholder"})
+                if placeholder_found:
+                    endpoint = "{}{}{}".format(APP['middleware']['base_url'], APP['middleware']['content_root_url'], org_id)
+                    response = middleware_api(APP, endpoint)
+                    pages = response['data']
+                    for page in pages:
+                        page_html = page['Description']['Html']
+                        if page['Type'] == 0 and page_html:
+                            soup_html = BeautifulSoup(page_html, 'html')
+                            placeholders = soup_html.find_all('p', attrs={"data-type": "placeholder"})
+                            updated = False
+                            for placeholder in placeholders:
+                                sakai_id = placeholder.attrs['data-sakaiid']
+                                file_name = os.path.basename(sakai_id)
```

Comment:
Add some conditional logic for different types of replacements: audio, video, youtube (see https://jira.cilt.uct.ac.za/browse/AMA-674?focusedCommentId=223598&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-223598)